text/metrics/0185-uptime-monitoring-semantic-conventions.md

```diff
@@ -0,0 +1,104 @@
+
+This use case can also be met with a restart count metric, but "up and to the right" uptime graphs are more comfortable for users.
+
+#### Alerting on restarts using uptime
+
+Some customers want to configure an alert whenever their process (re)starts. Typically, they do this by configuring an alert that fires when the absolute value of the uptime (time since process start) is below a certain threshold.
+
+This type of alert is considered bad practice because it is, by definition, non-actionable. By the time the uptime is reported as 0, the process has already started and is now running. The alert will self-close as soon as the process has been running for a few minutes.
+
+We recommend in general not configuring alerts on this type of metric, but it is a common practice.
+
+#### Alerting on current process health
+
+Some customers want to configure an alert when a process is unhealthy, or the number of healthy processes falls below a certain level. They can do this using a boolean metric exported from each task that reports if it is currently healthy.
+
+In general, we recommend that services prefer exposing more detailed metrics instead of boolean health. For example, a user should prefer an alert defined as "successful qps > 0.1" instead of an alert defined as "healthy = TRUE". The reason for this is that boolean health metrics essentially move the alert condition definition inside the process, removing the user's ability to control those conditions.
+
+#### Alerting on crashing processes
+
+Users want to know if their processes are restarting frequently (for example, as part of a crash loop). This is the hardest type of metric to report, because restarting processes may not be healthy enough to report that they are crashing.
+
+When there is some kind of supervisor process that is separate from the process being monitored, this can be tracked by having the supervisor process report the number of times a process has restarted. Then an alert can be configured when the rate of that count exceeds a threshold (e.g. "process has restarted ≥ 5 times in one hour").
+
+Since supervisor processes like this do not often exist, other ways to partially achieve this are to configure an alert on the average uptime over a long time period (e.g. "average uptime over 60m window < 15m"), and/or alert when a metric is absent.
+
+## Internal details
+
+We propose the following metrics be used to track uptime within OpenTelemetry:
+
+| Name                   | Description                  | Units | Instrument Type              |
+| ---------------------- | ---------------------------- | ----- | -----------------------------|
+| *.uptime               | Seconds since last restart   | s     | Asynchronous Gauge           |
```

Comment:
The [general recommendations](https://github.com/open-telemetry/opentelemetry-specification/tree/a25d5f03ab58ecf88c09f635df97d2328b5ba237/specification/metrics/semantic_conventions#instrument-naming) appear to say that `.time` suffix needs a dot. Should this be `*.up.time` if we follow the recommendation?

Reply:
It seems `*.time` is normally used for things that have additive property.